src/main/java/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,54 @@
+package bookstore.repository.impl;
+
+import bookstore.entity.Book;
+import bookstore.exception.EntityNotFoundException;
+import bookstore.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+@RequiredArgsConstructor
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Override
+    public Book save(Book book) {
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
```

Comment:
Try with resources isn't good idea here as session will be closed if something happened and rollback doesn't work in such case
Use try-catch-finally